doc/tutorials/01-lennard_jones/01-lennard_jones.ipynb

```diff
@@ -393,98 +441,103 @@
-    "r_bins = 100\n",
-    "r_min = 0.0\n",
-    "r_max = system.box_l[0] / 2.0\n",
-    "# Initialize RDF accumulator\n",
-    "rdf_obs = RDF(ids1=system.part[:].id, min_r=r_min, max_r=r_max, n_r_bins=r_bins)\n",
-    "rdf_acc = MeanVarianceCalculator(obs=rdf_obs, delta_N=sampling_interval)\n",
-    "# Accumulate results automatically during the integration\n",
-    "system.auto_update_accumulators.add(rdf_acc)\n",
-    "\n",
-    "# Take measurements\n",
-    "time = np.zeros(sampling_iterations)\n",
-    "instantaneous_temperature = np.zeros(sampling_iterations)\n",
-    "etotal = np.zeros(sampling_iterations)\n",
-    "\n",
-    "for i in range(1, sampling_iterations + 1):\n",
-    "    system.integrator.run(sampling_interval)\n",
-    "    # Measure energies\n",
-    "    energies = system.analysis.energy()\n",
-    "    kinetic_temperature = energies['kinetic'] / (1.5 * N_PART)\n",
-    "    etotal[i - 1] = energies['total']\n",
-    "    time[i - 1] = system.time\n",
-    "    instantaneous_temperature[i - 1] = kinetic_temperature\n",
-    "\n",
-    "# Finalize the correlator and obtain the results\n",
-    "msd_corr.finalize()\n",
-    "msd = msd_corr.result()\n",
-    "\n",
-    "# Finalize the accumulator and obtain the results\n",
-    "rdf = rdf_acc.get_mean()\n",
-    "r = rdf_obs.bin_centers()"
+    "F_TOL = 1e-2\n",
+    "E_TOL = 1e-5\n",
+    "DAMPING = 30\n",
+    "MAX_STEPS = 10000\n",
+    "MAX_DISPLACEMENT = 0.01 * LJ_SIG\n",
+    "EMSTEP = 10"
    ]
   },
   {
    "cell_type": "markdown",
-   "metadata": {},
+   "metadata": {
+    "solution2": "hidden",
+    "solution2_first": true
+   },
    "source": [
-    "We now use the plotting library <tt>matplotlib</tt> available in Python to visualize the measurements."
+    "**Exercise:**\n",
+    "\n",
+    "* Use <tt>[espressomd.integrate](http://espressomd.org/html/doc/espressomd.html#module-espressomd.minimize_energy).set_steepest_descent</tt> to relax the initial configuration.\n",
+    "  Use a maximal displacement to <tt>MAX_DISPLACEMENT</tt>.\n",
+    "  A damping constant <tt>gamma = DAMPING</tt> usually is a good choice.\n",
+    "* Use the relative change of the system total energy and the maximal force as a convergence criterion.\n",
+    "  See the [<tt>espressomd.analyze</tt>](http://espressomd.org/html/doc/espressomd.html#module-espressomd.analyze) documentation on how to obtain energies and the [<tt>espressomd.particle_data</tt> module](http://espressomd.org/html/doc/espressomd.html#module-espressomd.particle_data) for the forces.\n",
+    "  The steepest descent has converged if the relative force change < <tt>F_TOL</tt>\n",
+    "* Break the minimization loop after a maximal number of <tt>MAX_STEPS</tt> steps or if convergence is achieved.\n",
+    "  Check for convergence every <tt>EMSTEP</tt> steps.\n",
+    "  \n",
+    "***Hint:*** To obtain the initial fores one has to initialize integrator using <tt>integ_steps=0</tt>."
```

Comment:
I think `system.integrator.run(0)` has to be mentioned explicitly. I wouldn't have understood that otherwise, I think. 

Could you please make an attempt to simplify the loop? A single criterion such as absolute force should be enough, IMO.
There only is a point in counting the number of times the loop is executed, if the non-convergence case is actually treated.

Reply:
I'll implement the requested changes suggested below, only two comments on the minimization loop:
* I agree that this reads complicated. But the problem is that in steepest descent we minimize energies whereas the MD relies on forces only, so for me the general question is how to define a reasonable convergence criterion? I agree that this reads slightly complicated but only on the python side (unrelated to ESPResSo). At the same time this follows the practice of other MD packages (Gromacs, LAMMPS, ...). If there is any favour on which criterion is better in the ESPResSo community I'm happy to change accordingly.
* the non-convergence case should actually be treated.
I strongly believe that the adapted steepest descent with above convergence criteria is something we should offer to the user in the mid-term, which would also greatly simplify the minimization in the code (see discussion in #3898).

Reply:
The non-convergence case **is** actually treated and I added more explanations to the `system.integrator.run(0)` command.